app/segments/drill/[id]/submission/input.js

```diff
@@ -137,6 +141,136 @@ async function uploadAttempt(
 }
 
 //TODO: Create a function to check leaderboard and update if needed
+function checkLeaderboardUpdate(uploadData, drillInfo, currentLeaderboard) {
+  const leaderboardData = currentLeaderboard["data"];
+
+  //used if an attempt already exists
+  const currentBest =
+    leaderboardData[uploadData.uid]?.[drillInfo.mainOutputAttempt]?.value;
+
+  const newAttempt = {
+    [drillInfo.mainOutputAttempt]: {
+      id: uploadData.id,
+      value: uploadData[drillInfo.mainOutputAttempt],
+    },
+  };
+
+  console.log("currentBest: ", currentBest);
+
+  //check if the user exists on the leaderboard
+  if (leaderboardData[uploadData.uid] == undefined) {
+    console.log("User not on leaderboard, uploading this attempt");
+
+    uploadNewLeaderboard(
+      leaderboardData,
+      newAttempt,
+      uploadData.uid,
+      uploadData.did,
+    );
+  } else {
+    //Determine if lower is better for mainOutputAttempt
+    if (drillInfo.aggOutputs[drillInfo.mainOutputAttempt].lowerIsBetter) {
```

Comment:
const lowerIsBetter = drillInfo.aggOutputs[drillInfo.mainOutputAttempt].lowerIsBetter
const isNewAttemptBest = lowerIsBetter ? uploadData[drillInfo.mainOutputAttempt] < currentBest : uploadData[drillInfo.mainOutputAttempt] > currentBest
if(isNewAttemptBest){upload} else {console.log}

Reply:
This is cool, hard to wrap my head around it initially but I am gonna try to use this more when I', coding